src/metabase/query_processor/middleware/fetch_source_query.clj

```diff
@@ -109,6 +109,26 @@
          (assoc m :field_ref [:field (:name m) {:base-type (:base_type m)}]))
        metadata))
 
```

Comment:
Lifted this out as a private function for better readability as `card-id->source-query-and-metadata` was pretty big.